packages/apps/shopify-api/lib/billing/create-usage-record.ts

```diff
@@ -0,0 +1,172 @@
+import {ConfigInterface} from '../base-types';
+import {BillingError, GraphqlQueryError} from '../error';
+import {GraphqlClient, graphqlClientClass} from '../clients/admin';
+
+import {
+  AppSubscription,
+  BillingCreateUsageRecord,
+  BillingCreateUsageRecordParams,
+  UsageRecord,
+  UsageRecordCreateResponse,
+  Money,
+} from './types';
+import {assessPayments} from './check';
+
+interface InternalParams {
+  client: GraphqlClient;
+  isTest?: boolean;
+}
+
+interface CreateUsageRecordVariables {
+  description: string;
+  price: Money;
+  subscriptionLineItemId: string;
+  idempotencyKey?: string;
+}
+
+const CREATE_USAGE_RECORD_MUTATION = `
+mutation appUsageRecordCreate($description: String!, $price: MoneyInput!, $subscriptionLineItemId: ID!) {
```

Comment:
Should we name this in a more "unique" way for tracking purposes? Could be a follow-up if we're not in other methods.